include/envoy/grpc/async_client_manager.h

```diff
@@ -29,6 +29,12 @@ class AsyncClientManager {
 public:
   virtual ~AsyncClientManager() = default;
 
+  virtual RawAsyncClientSharedPtr
+  getOrCreateRawAsyncClient(const envoy::config::core::v3::GrpcService& grpc_service,
+                            Stats::Scope& scope, bool skip_cluster_check) {
+    return factoryForGrpcService(grpc_service, scope, skip_cluster_check)->create();
```

Comment:
I think you can make this method PURE and move this default implementation into the mock in test/mocks/grpc/mocks.h. It's a little better IMO to have this class be 100% pure rather than 50% pure.

I did some grepping around the codebase and I think there are just 2 impls of this interface: the mock and the real one, and the real one has its own impl of this method.

Reply:
Thanks, will fix.